client/extensions/woocommerce/components/query-settings-products/index.js

```diff
@@ -0,0 +1,39 @@
+/**
+ * External dependencies
+ *
+ * @format
+ */
+
+import { Component } from 'react';
+import { PropTypes } from 'prop-types';
+import { connect } from 'react-redux';
+
+/**
+ * Internal dependencies
+ */
+import { fetchSettingsProducts } from 'woocommerce/state/sites/settings/products/actions';
+
+class QuerySettingsProducts extends Component {
+	static propTypes = {
+		siteId: PropTypes.number.isRequired,
+		fetchSettingsProducts: PropTypes.func.isRequired,
+	};
+
+	componentDidMount() {
+		this.props.fetchSettingsProducts( this.props.siteId );
+	}
+
+	componentWillReceiveProps( nextProps ) {
+		if ( this.props.siteId === nextProps.siteId ) {
+			return;
+		}
+		this.request( nextProps );
```

Comment:
Won't this throw an error, I'm not seeing `request` defined in this component.

Reply:
Scarry but it does not. I wonder why. It should not be here ofc - but the console stays clean. Removing.